src/main/java/seedu/address/model/assignment/Status.java

```diff
@@ -2,12 +2,20 @@
 
 import static java.util.Objects.requireNonNull;
 
-public class Status {
+public class Status implements Comparable<Status> {
     public static final String MESSAGE_CONSTRAINTS = "Status of assignment should be clearly defined!";
     public final String value;
 
     enum StatusType {
-        COMPLETED, LATE, PENDING
+        COMPLETED(2),
+        LATE(0),
```

Comment:
Nice that you have considered to put late assignments all the way at the top of the list